consul/resource_consul_service.go

```diff
@@ -497,11 +497,18 @@ func retrieveService(client *consulapi.Client, name string, ident string, node s
 	// Only one service with a given ID may be present per node
 	for _, s := range services {
 		if (s.ServiceID == ident) && (s.Node == node) {
+			// Fetch health-checks for this service
 			healthChecks, _, err := client.Health().Checks(name, &qOpts)
 			if err != nil {
 				return nil, fmt.Errorf("Failed to fetch health-checks: %v", err)
 			}
-			s.Checks = healthChecks
```

Comment:
Could this cause backwards incompatibility issues? By returning less checks than expected?

Reply:
It's true that there will be less health-checks but the previous behaviour was returning health-checks not associated with this service instance, I think it was a bug, people relying on this could not make the correct decisions could they?